TACC/Frontera-Portal
# Task/fp 621  fp 662 common section component

## Overview: ##

Create and use `Section` component for generic section layout.

## PR Status: ##


## Related Jira tickets: ##

* some of [FP-385](https://jira.tacc.utexas.edu/browse/FP-385)
    * [FP-753](https://jira.tacc.utexas.edu/browse/FP-753)
    * [FP-662](https://jira.tacc.utexas.edu/browse/FP-662)
* some of [FP-722](https://jira.tacc.utexas.edu/browse/FP-722)

## Summary of Changes: ##

- Create `_common/Section` component.
- Use `_common/Section` component for each section.
- Remove and Extract styles and markup after `_common/Section`.
- Make sidebar width and icon left alignment consistent.
- Add comments.

## Testing Steps: ##
1. Load all sections.
2. Ensure headers and header element alignments are consistent.
3. Ensure space around sections is consistent.
4. Ensure header font size is consistent.

### Known Issues: ###
- Scrolling broken on panels within content. _This will be fixed by creation of a `_common/SectionContent` component._

## UI Photos:
![1 Dashboard](https://user-images.githubusercontent.com/62723358/95544763-b54d1d00-09c1-11eb-9393-7d4e504bbd7c.png)
![2 Data Files](https://user-images.githubusercontent.com/62723358/95544764-b7af7700-09c1-11eb-88d8-a62af2836e11.png)
![3 Allocations](https://user-images.githubusercontent.com/62723358/95544765-b8480d80-09c1-11eb-947e-06cafc1dcc27.png)
![3 Applications - Form](https://user-images.githubusercontent.com/62723358/95544768-b8480d80-09c1-11eb-94c2-aa06e2e5e7d6.png)
![3 Applications - Launcher](https://user-images.githubusercontent.com/62723358/95544771-b8e0a400-09c1-11eb-9e2c-9a448a8183f6.png)
![3 Applications - None](https://user-images.githubusercontent.com/62723358/95544772-b8e0a400-09c1-11eb-9685-0ff4030a4b86.png)
![4 History](https://user-images.githubusercontent.com/62723358/95544773-b9793a80-09c1-11eb-84fc-9e754e6bcfe1.png)
![5 UI Patterns - Section Maximum Props](https://user-images.githubusercontent.com/62723358/95544775-b9793a80-09c1-11eb-8280-a230fdec8f17.png)
![5 UI Patterns - Section Minimal Props](https://user-images.githubusercontent.com/62723358/95544776-b9793a80-09c1-11eb-988b-d587458f8011.png)
![5 UI Patterns - Section No Message - No Shrink](https://user-images.githubusercontent.com/62723358/95544777-ba11d100-09c1-11eb-845e-17d4f3b7547e.png)
![5 UI Patterns - Section No Message - Shrink to Scroll](https://user-images.githubusercontent.com/62723358/95544778-ba11d100-09c1-11eb-97aa-d552fd42140e.png)

## Notes: ##
